internal/flipflop/handler.go

```diff
@@ -132,11 +132,18 @@ func (cth *ConditionTaskHandler) Run(ctx context.Context) error {
 	case rctypes.PowerCycleBMC:
 		return cth.powerCycleBMC(ctx)
 	case rctypes.SetPowerState:
-		return cth.setPowerState(ctx)
+		return cth.setPowerState(ctx, cth.task.Parameters.ActionParameter)
```

Comment:
cth already gives setPowerState() access to the task. No need to also pass it in as a param